include/spi.h

```diff
@@ -218,6 +228,13 @@ typedef int (*spi_api_io_async)(struct device *dev,
 typedef int (*spi_api_release)(struct device *dev,
 			       const struct spi_config *config);
 
+/**
+ * @typedef spi_api_trigger
+ * @brief Callback API for triggering SPI device.
+ * See spi_trigger() for argument descriptions
+ */
+typedef int (*spi_api_trigger)(struct device *dev,
```

Comment:
What would be the uses cases for this?

What would require one to pre-load a spi transaction, and wait for a signal to actually start it, instead of waiting for that signal to do the transaction as usual?